cdc/sink/common/flow_control.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"log"
+	"sync"
+	"sync/atomic"
+
+	"github.com/edwingeng/deque"
+	"github.com/pingcap/errors"
+	cerrors "github.com/pingcap/ticdc/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// TableMemorySizeController is designed to curb the total memory consumption of processing
+// the event streams in a table.
+// A higher-level controller more suitable for direct use by the processor is TableFlowController.
+type TableMemorySizeController struct {
```

Comment:
```suggestion
type TableMemoryQuota struct {
```

It feels like a memory quota.